HumanExposure/factotum
# 1506 datadocument soft delete

Closes #1506, which prevents the associate datadocument file from being deleted when the datadocument record is deleted. 

Django doesn't delete associated files by itself, so at some point we introduced the django-cleanup module to handle this. I eventually figured out that we just needed to apply an @ignore decorator to the DataDocument model definition to prevent django-cleanup from deleting these files.

To be safe, I wanted to test not only that datdocument files get preserved when datadocuments are deleted, but also that product image files do NOT get preserved when products are deleted. (Unfortunately I'm still having a test failure, in that the product file isn't getting deleted as expected in the test. If this failure occurs for the reviewer, we might need to comment that line out for the time being. The functionality can still be tested manually.)

To test manually in a dev environment: 

1. Paste a pdf file into the /media folder.
2. Using MySQL Workbench or similar, choose a record in the dashboard_datadocument table and update the 'file' value to be the name of the pdf file.
3. Go to the factotum page for that datadocument and verify that you can click the pdf icon and load the file as expected
4. Delete the datadocument, and verify that the pdf file still exists in the /media folder

Then repeat the same basic procedure with the product table and its 'image' value. Unlike the datadocument file, the product image should be deleted when the product is deleted.
